### ISSUE TITLE
GitHub inline tipping support

### ISSUE THREAD

**Issue by jdkuki** [brave/brave-browser#5040]
## Description 

Feature request: 
GitHub inline tipping should be enabled in several aspects of the GitHub website. 
See screenshots/below specification document for affected ares. 

## Supported Areas:
```
https://github.com/<user>/<repository>/issues/<issue_number>
https://github.com/<user>/<repository>/pull/<pr_number>
https://github.com/<user>?tab=stars
https://github.com/<user>/<repository>/commits/<branch_name>
https://github.com/orgs/<org>/people
https://github.com/orgs/<org>/teams/<team_name>/members
https://gist.github.com/<user>/<gist_number>
```

## Steps to Reproduce

   1. Join Rewards
   2. Browse to user profile (ex. github.com/jdkuki)
   3. Click "stars tab"
   4. Tip buttons should appear next to star button
   5. Click tip button
   6. Tip banner should appear with text "(profile_name)* on GitHub". Name should correspond to creator
        of starred item
   7. Select tip amount and send
   8. Transaction should show up in pending contributions or balance should update according to 
       publisher verification status
   9. Navigate to brave://rewards
   10. Under tips settings, uncheck "GitHub"
   11. Open pages with injected tip buttons should see tip buttons disappear.

   *If profile name is not set, banner will display username handle instead

## Screenshots
Please reference specification for exact appearance. Few examples:
**PR**
![image](https://user-images.githubusercontent.com/8422122/60373216-4ad2c200-99b4-11e9-8a79-0a151d635fa0.png)

**Issues**
![image](https://user-images.githubusercontent.com/8422122/60373531-58d51280-99b5-11e9-9903-d17f390cd651.png)

**Commits**
![image](https://user-images.githubusercontent.com/8422122/60373262-6ccc4480-99b4-11e9-9172-b74b48afb345.png)

## Actual result:

n/a

## Version/Channel Information:

- Can you reproduce this issue with the current release?
- Can you reproduce this issue with the beta channel?
- Can you reproduce this issue with the dev channel?
- Can you reproduce this issue with the nightly channel?

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields?
- Does the issue resolve itself when disabling Brave Rewards?
- Is the issue reproducible on the latest version of Chrome?

**Comment by LaurenWags** [comment]
Verified passed with

Brave | 0.69.130 Chromium: 77.0.3865.90 (Official Build) (64-bit)
-- | --
Revision | 58c425ba843df2918d9d4b409331972646c393dd-refs/branch-heads/3865@{#830}
OS | macOS Version 10.13.6 (Build 17G5019)

* Verified STR from description for each of the affected areas (also listed in the description)
* Encountered https://github.com/brave/brave-browser/issues/5562
* Verified tip button on `https://github.com/<org>/<repository>/issues/<issue_number>`
<img width="1193" alt="Screen Shot 2019-10-01 at 12 54 23 PM" src="https://user-images.githubusercontent.com/28145373/65991610-a00eeb80-e45b-11e9-82cb-9ea23dd1aaef.png">
* Verified tip button on PR
<img width="1189" alt="Screen Shot 2019-10-01 at 12 57 10 PM" src="https://user-images.githubusercontent.com/28145373/65991666-b321bb80-e45b-11e9-82d0-eb740b313ede.png">
* Verified tip button on stars tab of user profile
<img width="1193" alt="Screen Shot 2019-10-01 at 2 57 31 PM" src="https://user-images.githubusercontent.com/28145373/65991726-d3517a80-e45b-11e9-80d5-b70e74e958b0.png">
* Verified tip button on commits > branch name
<img width="1194" alt="Screen Shot 2019-10-01 at 2 48 50 PM" src="https://user-images.githubusercontent.com/28145373/65991775-ecf2c200-e45b-11e9-9b3d-f9c1b3de26db.png">
* Verified tip button on people page of org
<img width="1187" alt="Screen Shot 2019-10-01 at 2 49 41 PM" src="https://user-images.githubusercontent.com/28145373/65991821-fc720b00-e45b-11e9-9949-3b22ce5f4a19.png">
* Verified tip button on team members page
<img width="1192" alt="Screen Shot 2019-10-01 at 2 52 43 PM" src="https://user-images.githubusercontent.com/28145373/65991849-0bf15400-e45c-11e9-93d6-4278456fe98b.png">
* Verified tip button on gists
<img width="1196" alt="Screen Shot 2019-10-01 at 2 54 06 PM" src="https://user-images.githubusercontent.com/28145373/65991882-19a6d980-e45c-11e9-93c6-40c14c7a8f29.png">
* Verified tip banner showed `profile name on GitHub` if existed and `username on GitHub` if it did not
<img width="1199" alt="Screen Shot 2019-10-01 at 2 50 02 PM" src="https://user-images.githubusercontent.com/28145373/65991967-43f89700-e45c-11e9-8c41-57c1ee32889e.png">
<img width="1191" alt="Screen Shot 2019-10-01 at 2 51 17 PM" src="https://user-images.githubusercontent.com/28145373/65991972-465af100-e45c-11e9-95b8-0ddc2599daed.png">
* Verified tips showed in panel or pending where appropriate (based on established states):
<img width="642" alt="Screen Shot 2019-10-01 at 3 01 25 PM" src="https://user-images.githubusercontent.com/28145373/65992021-5ffc3880-e45c-11e9-87cc-145c619f242b.png">
<img width="947" alt="Screen Shot 2019-10-01 at 3 01 33 PM" src="https://user-images.githubusercontent.com/28145373/65992024-612d6580-e45c-11e9-836c-ea11eb784501.png">

**Comment by forresthopkinsa** [comment]
IMO it's *essential* that we have the option to tell Brave to publish the tip on the relevant issue / PR / commit.

It's just not rewarding at all to send someone a tip and not know if they'll ever receive it – additionally, I want them to know who the tip is from and **why** it was sent to them. (Anecdotally, I installed the Brave browser and registered an Uphold account purely because I want to reward various developers for fixing bugs I reported)

It might be silly, but it is such a critical factor for the *human* aspect of tipping that I'm tempted to use Gitcoin's [system](https://youtu.be/uE1ukhdhKWk?t=71) instead, even though it's so much more clunky.

Potato quality example of what I'm talking about, taken from the above video:

![image](https://user-images.githubusercontent.com/4433153/102672143-4cb2b700-414d-11eb-8a98-2fbfd2d66f0d.png)
